pymsis/utils.py

```diff
@@ -208,6 +207,12 @@ def get_f107_ap(dates: npt.ArrayLike) -> tuple[npt.NDArray, npt.NDArray, npt.NDA
     """
     dates = np.asarray(dates, dtype=np.datetime64)
     data = _DATA or _load_f107_ap_data()
+    if dates[-1] > data["dates"][~np.repeat(data["warn_data"], 8)][-1]:
```

Comment:
If someone didn't sort their inputs/request, we should use a max here I think. Could you also add a quick comment about what this if statement is doing.
```suggestion
    # If our requested data time is after the cached values we have, go and download a new file
    # to refresh the local file cache
    if dates.max() > data["dates"][~np.repeat(data["warn_data"], 8)][-1]:
```

Reply:
Thanks @greglucas, I've commented the date comparison check